File: Cap 09/Topico 9.4/Topico 9.4.1 a 9.4.3 juntos/regmatrix.swift
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 9 - Topicos 9.4.1 a 9.4.3
 * 
 * Programa ..: MANIPULACAO DE REGISTRO DE MATRIZ

 */

import Foundation

typealias Bimestre = [Float]

struct Aluno {
    var nome: String
    var turma: Character
    var sala: Int
    var notas: Bimestre
}

var alunos: [Aluno] = []

print("REGISTRO ESCOLAR - MATRICIAL (8 ALUNOS)")
print()

for i in 1...8 {
    print("ALUNO \(i)")
    
    print("Entre o nome ......: ", terminator: "")
    let nome = readLine() ?? ""
    
    print("Entre a turma .....: ", terminator: "")
    let turma = readLine()?.first ?? " "
    
    print("Entre a sala ......: ", terminator: "")
    let salaStr = readLine() ?? ""
    let sala = Int(salaStr) ?? 0
    
    var notas: Bimestre = []
    for j in 1...4 {
        print("Entre a \(j)a. nota ..: ", terminator: "")
        let notaStr = readLine() ?? ""
        let nota = Float(notaStr) ?? 0.0
        notas.append(nota)
    }
    
    alunos.append(Aluno(nome: nome, turma: turma, sala: sala, notas: notas))
    print()
}

print("DADOS DOS ALUNOS")
print("Aluno  Nome                           Sala  Nota1 Nota2 Nota3 Nota4")
print("-----  ------------------------------ ----  ----- ----- ----- -----")

for (i, aluno) in alunos.enumerated() {
    let alunoIndex = i + 1
    let nome = String(aluno.nome.prefix(30)).padding(toLength: 30, withPad: " ", startingAt: 0)
    let sala = String(format: "%3d", aluno.sala)
    let notas = aluno.notas.map { String(format: "%5.1f", $0) }.joined(separator: " ")
    
    print(String(format: "%5d  %@  %@  %@", alunoIndex, nome, sala, notas))
}

File: Cap 09/Topico 9.4/Topico 9.4.1 a 9.4.3 juntos/regmatrix.js
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 9 - Topicos 9.4.1 a 9.4.3
 * 
 * Programa ..: MANIPULACAO DE REGISTRO DE MATRIZ

 */

const readline = require('readline-sync');

const ALUNO = new Array(8);
const CAD_ALUNO = {
  NOME: "",
  TURMA: "",
  SALA: 0,
  NOTAS: []
};

let I, J;
console.log("REGISTRO ESCOLAR - MATRICIAL (8 ALUNOS)\n");

for (I = 0; I < 8; I++) {
	
  console.log(`ALUNO ${I + 1}`);
  
  process.stdout.write("Entre o nome ......: ");
  ALUNO[I] = Object.assign({}, CAD_ALUNO);
  ALUNO[I].NOME = readline.question();
  
  process.stdout.write("Entre a turma .....: ");
  ALUNO[I].TURMA = readline.question()[0];
  
  process.stdout.write("Entre a sala ......: ");
  ALUNO[I].SALA = parseInt(readline.question());
  
  ALUNO[I].NOTAS = [];
  for (J = 0; J < 4; J++) {
    process.stdout.write(`Entre a ${J + 1}a. nota ..: `);
    ALUNO[I].NOTAS.push(parseFloat(readline.question()));
  }
  console.log();
}
console.log("DADOS DOS ALUNOS");
console.log("Aluno Nome                             Sala Nota1 Nota2 Nota3 Nota4");
console.log("----- -------------------------------- ---- ----- ----- ----- -----");
for (I = 0; I < 8; I++) {
  console.log(
    `${(I + 1).toString().padStart(5)} ` +
    `${ALUNO[I].NOME.padEnd(30)} ` +
    `${ALUNO[I].SALA.toString().padStart(6)} ` +
    `${ALUNO[I].NOTAS[0].toFixed(1).padStart(5)} ` +
    `${ALUNO[I].NOTAS[1].toFixed(1).padStart(5)} ` +
    `${ALUNO[I].NOTAS[2].toFixed(1).padStart(5)} ` +
    `${ALUNO[I].NOTAS[3].toFixed(1).padStart(5)}`
  );
}
